pkg/controller/instance/instance_controller.go

```diff
@@ -350,12 +367,27 @@ func createPlan(mgr manager.Manager, planName string, instance *kudov1alpha1.Ins
 		return err
 	}
 
-	if err := mgr.GetClient().Create(context.TODO(), &planExecution); err != nil {
+	if err := mgr.GetClient().Create(ctx, &planExecution); err != nil {
 		log.Printf("InstanceController: Error creating planexecution \"%v\": %v", planExecution.Name, err)
 		recorder.Event(instance, "Warning", "CreatePlanExecution", fmt.Sprintf("Error creating planexecution \"%v\": %v", planExecution.Name, err))
 		return err
 	}
 	recorder.Event(instance, "Normal", "PlanCreated", fmt.Sprintf("PlanExecution \"%v\" created", planExecution.Name))
+
+	//update the instance with plan
+	active := &instance.Status.ActivePlan
```

Comment:
Yeah I think doing it here makes more sense to me than inside planexecutioncontroller 👏